src/routes/profile.svelte

```diff
@@ -3,21 +3,39 @@
   import isoForURL from '@/utils/iso-for-url.js';
 
   export async function preload(page, session) {
+    let requestedEmail = page.query.user;
     let timelineFetchedUntil = new Date();
     timelineFetchedUntil.setMonth(timelineFetchedUntil.getMonth() - 3);
+    const { currentUser } = await getInitialData(this, session, new Map([
+      ['currentUser', '/account'],
+    ]));
+    if (currentUser == null) {
+      this.error(403, 'Profile');
+    }
+    const isMe = requestedEmail == currentUser.email;
+    const allowed = requestedEmail == null || isMe || currentUser.is_admin;
+    // non-admin requested a different profile
+    if (!allowed) {
+      this.redirect(302, `/profile`);
+    }
+    if (requestedEmail == null) {
+      requestedEmail = currentUser.email;
+    }
+
     const {
       account,
       ...initialData
     } = await getInitialData(this, session, new Map([
-      ['account', '/account'],
-      ['timeline', `/account/timeline?start_date=${isoForURL(timelineFetchedUntil)}`],
-      ['statistics', `/account/statistics?start_date=${isoForURL(timelineFetchedUntil)}`],
-      ['notificationSettings', '/account/notification_settings'],
+      ['account', `/accounts/${requestedEmail}`],
```

Comment:
Why are you requesting /account for the second time? (first at line 9)

Reply:
They don't have to be the same account. The one requested here is of the profile we are viewing rather than the logged-in user. I understand that it is redundant in cases where a user is viewing their own profile, but the code would be a bit messier to handle that case.